tests/test_ingestion.py

```diff
@@ -0,0 +1,183 @@
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import absolute_import, division, print_function
+
+import os
+import shutil
+import tempfile
+import unittest
+
+import lsst.utils.tests
+import lsst.pex.exceptions as pexExcept
+import lsst.daf.persistence as dafPersist
+import lsst.ap.verify.ingestion as ingestion
+
+
+# TODO: convert test data to obs_test in DM-13849
+# In the meantime, run scons -j or pytest --forked
+class IngestionTestSuite(lsst.utils.tests.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            cls.testData = lsst.utils.getPackageDir("testdata_decam")
+        except pexExcept.NotFoundError:
+            message = "testdata_decam not setup. Skipping."
+            raise unittest.SkipTest(message)
+
+        cls.testApVerifyData = os.path.join('tests', 'ingestion')
+        cls.rawDataId = {'visit': 229388, 'ccdnum': 1}
+        # TODO: butler queries fail without this extra info; related to DM-12672?
+        cls.calibDataId = {'visit': 229388, 'ccdnum': 1, 'filter': 'z', 'date': '2013-09-01'}
+        cls.defectDataId = {'path': os.path.join('defects', 'D_n20150105t0115_c23_r2134p01_bpm.fits')}
+
+    def setUp(self):
+        self._repo = tempfile.mkdtemp()
+        self._calibRepo = os.path.join(self._repo, 'calibs')
+        templateRepo = os.path.join(IngestionTestSuite.testApVerifyData, 'repoTemplate')
+
+        # Initialize as a valid repository
+        # Can't call copytree on (templateRepo, self._repo) because latter already exists
+        testFiles = os.listdir(templateRepo)
+        for testFile in testFiles:
+            original = os.path.join(templateRepo, testFile)
+            copy = os.path.join(self._repo, testFile)
+            if os.path.isdir(original):
+                shutil.copytree(original, copy)
+            else:
+                shutil.copy2(original, copy)
+
+        # Initialize calib repo
+        # Making the directory appears to be both necessary and sufficient
+        os.mkdir(self._calibRepo)
+
+    def _rawButler(self):
+        """Return multiple ways of querying calibration repositories.
+
+        Returns
+        -------
+        butler : `lsst.daf.persistence.Butler`
+            A butler that should all be capable of finding ingested science data.
+        """
+        return dafPersist.Butler(inputs={'root': self._repo, 'mode': 'r'})
+
+    def _calibButler(self):
+        """Return multiple ways of querying calibration repositories.
+
+        Returns
+        -------
+        butlers : `lsst.daf.persistence.Butler`
+            A butler that should all be capable of finding ingested calibration data.
+        """
+        return dafPersist.Butler(inputs={
+            'root': self._repo,
+            'mode': 'r',
+            'mapperArgs': {'calibRoot': self._calibRepo}})
+
+    def tearDown(self):
+        shutil.rmtree(self._repo, ignore_errors=True)
```

Comment:
This does make sense here but I'm just laughing to myself about `ignore_errors` as an argument 😆